### PR TITLE
Afterjqueryedit

### PR DESCRIPTION
**Pull request by HrisikeshSinghaChowdhury**
Adding Edit Feature jquery...

Note: On double clicking the Show/Edit/Add the AJAX requests are sent multiple times.not handled this case

### CODE REVIEW

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@ def index
   # GET /microposts/1 or /microposts/1.json
   def show
     @micropost =  Micropost.find(params[:id])
+    @micropost.content
```
Comment (Ramyani-QI) [inline app/controllers/microposts_controller.rb]: @HrisikeshSinghaChowdhury why do we need this line?

File: app/views/microposts/new.js.erb [hunk 2]
```diff
@@ -1,4 +1,2 @@
-$(document).ready(function(){
- alert("Hii please proceed..");
+alert("Hii please proceed..");
  $("#blogs").append("<%= j render 'form', micropost: @micropost %>");
```
Comment (Ramyani-QI) [inline app/views/microposts/new.js.erb]: alignment is not correct!

File: app/views/microposts/edit.html.erb [hunk 3]
```diff
@@ -1,32 +1,8 @@
...
-             <%= link_to 'Back', user_path(current_user), class: 'btn btn-secondary'%>
-
-            </div>
-        </div>
-      </div>
-
-    </div>
-  </section>
-<% end %>
+<%= render 'form', user: @user %>
```
Comment (Ramyani-QI) [inline app/views/microposts/edit.html.erb:3]: @HrisikeshSinghaChowdhury why user: @user !
where is it being used?

Comment (Ramyani-QI) [inline app/views/microposts/edit.html.erb]: @HrisikeshSinghaChowdhury why `user: @user ` ?
where is it being used?

Comment (HrisikeshSinghaChowdhury) [reply app/views/microposts/edit.html.erb]: sorry that was a copy paste mistake....

File: app/views/microposts/update.js.erb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+alert("Hurray Blog has been successfully edited....");
+$("#blog-edit").html("<%=@micropost.content%>");
```
Comment (Ramyani-QI) [inline app/views/microposts/update.js.erb]: @HrisikeshSinghaChowdhury how is it working?  how will it find the specific blog row for which you are updating the content?

File: app/views/microposts/_micropost.html.erb [hunk 5]
```diff
@@ -1,10 +1,10 @@
 <tr>
-            <td>
+            <td id="blog-edit">
```
Comment (Ramyani-QI) [inline app/views/microposts/_micropost.html.erb]: @HrisikeshSinghaChowdhury single id can not be present multiple times in a single page. this partial is being called for every micropost in a loop. you need to set specific class (eg: "blog-<%= micropost.id %>") to that row and also not in `<td>` it should be is `<tr>` itself and full tr should be replaced in update.js.erb while updating the micropost

Comment (HrisikeshSinghaChowdhury) [reply app/views/microposts/_micropost.html.erb]: ok understood

File: app/views/microposts/update.js.erb (hunk 4 above)
Comment (HrisikeshSinghaChowdhury) [reply app/views/microposts/update.js.erb]: it's working only first time

File: app/views/microposts/edit.html.erb [hunk 6]
```diff
@@ -1,32 +1,8 @@
...
-             <%= link_to 'Back', user_path(current_user), class: 'btn btn-secondary'%>
-
-            </div>
-        </div>
-      </div>
-
-    </div>
-  </section>
-<% end %>
+<%= render 'form', micropost: @micropost %>
```
Comment (Ramyani-QI) [inline app/views/microposts/edit.html.erb:3]: @HrisikeshSinghaChowdhury  why do we need `micropost: @micropost` ? where is it used?

Comment (HrisikeshSinghaChowdhury) [reply app/views/microposts/edit.html.erb:3]: It's not needed here because we are not passing any parameter as such to form partials

Comment (HrisikeshSinghaChowdhury) [reply app/views/microposts/edit.html.erb:3]: @Ramyani-QI As in the 'form partials we are directly attaching the form id with model microposts.....so we donnot need to pass and use any extra parameter during rendering.....
I may be wrong

File: app/views/microposts/_form.html.erb [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <%= provide(:title, 'Activity-Page') %>
-<%= form_with(model: @micropost, local: false) do |form| %>
+<%= form_with(model: @item, local: false) do |form| %>
```
Comment (Ramyani-QI) [inline app/views/microposts/_form.html.erb]: @HrisikeshSinghaChowdhury I gave you task to use `@micropost` with `item`

File: app/views/microposts/new.js.erb [hunk 8]
```diff
@@ -1,3 +1,2 @@
 alert("Hii please proceed..");
-$("#blogs").append("<%=j render 'form'%>");
-
+$("#blogs").append("<%=j render 'form', item: @micropost%>");
```
Comment (Ramyani-QI) [inline app/views/microposts/new.js.erb]: space missing before `%>`

**Pull request merged**